src/main/java/step2/LottoMachine.java

```diff
@@ -0,0 +1,59 @@
+package step2;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class LottoMachine {
+    private static final int LOTTO_TICKET_NUMBER_COUNT = 6;
+    private static final int COMPARE_SAME_NUMBER_COUNT = 1;
+    private LottoNumbers lottoNumbers = new LottoNumbers();
+
+    public List<LottoTicket> extractLottoTicketsByCount(int lottoCount) {
+        List<LottoTicket> lottoTicketList = new ArrayList<>();
+        for (int i = 0; i < lottoCount; i++) {
+            lottoTicketList.add(extractLottoTicket());
+        }
+
+        return lottoTicketList;
+    }
+
+    public List<Integer> checkLottoWinning(LottoTicket winningNumberTicket, List<LottoTicket> lottoTicketList) {
+        return lottoTicketList.stream().map(lottoTicket -> compareLottoNumber(winningNumberTicket, lottoTicket)).collect(Collectors.toList());
+    }
+
+    private int compareLottoNumber(LottoTicket winningNumberTicket, LottoTicket lottoTicket) {
+        List<Integer> myLottoNumbers = lottoTicket.getNumbers();
+        List<Integer> winningNumbers = winningNumberTicket.getNumbers();
+
+        return (int) winningNumbers.stream().filter(winningNumber
+                -> compareLottoNumber(winningNumber, myLottoNumbers)).count();
+    }
+
+    private boolean compareLottoNumber(int winningNumber, List<Integer> myLottoNumbers) {
+        return COMPARE_SAME_NUMBER_COUNT == myLottoNumbers.stream().filter(myLottoNumber
+                                                -> myLottoNumber == winningNumber).count();
```

Comment:
myLottoNumbers에 winningNumber가 포함되어있는 지 체크하는 로직이라면
```suggestion
        return myLottoNumbers.contains(winningNumber);
```
이렇게 사용할 수 있습니다:)

Reply:
스트림을 계속 쓰다보니까 그거에만 집중했나보네요 감사합니다!